lnrpc/rpc.proto

```diff
@@ -1009,6 +1009,16 @@ message HTLC {
 
     // Index identifying the htlc on the channel.
     uint64 htlc_index = 5;
+
+    // If this HTLC is involved in a forwarding operation, this field indicates
+    // the forwarding channel. For an outgoing htlc, it is the incoming channel.
+    // For an incoming htlc, it is the outgoing channel. When the htlc
+    // originates from this node or this node is the final destination,
+    // forwarding_channel will be zero.
+    uint64 forwarding_channel = 6;
```

Comment:
nit: since we can have a nil outgoing circuit, is it possible that we have a forwarded htlc that will have a 0 forwarding channel (temporarily)? Not the most familiar with that part of the switch, if possible perhaps update the comment so that people don't rely on 0 value meaning own payment.

Reply:
Comment added. I considered making this state explicit, but I don't think it is necessary. Htlcs that need to be forwarded, but haven't even reached the circuit map will also be reported with a zero `forwarding_channel`.